combat-trainer.lic

```diff
@@ -2687,43 +2687,45 @@ class TrainerProcess
   end
 
   def pray_mat(game_state)
-    unless @prayer_mat && @theurgy_supply_container
-      message '*** Cannot run PrayerMat; need prayer_mat and theurgy_supply_container defined'
+    ready_to_mat = (
+      DRCI.exists?(@prayer_mat, @prayer_mat_container) &&
+      DRCI.exists?("wine", @theurgy_supply_container)
+      )
+
+    unless ready_to_mat
+      DRC.message("*** Cannot run PrayerMat; need prayer mat in prayer_mat_container or in theurgy_supply_container, and wine in theurgy_supply_container")
       @training_abilities.delete('PrayerMat')
       return
     end
     waitrt?
-    retreat
+    DRC.retreat
     @equipment_manager.stow_weapon(game_state.weapon_name)
 
-    unless bput(
-      "get #{@prayer_mat} from my #{@prayer_mat_container}",
-      'You get', 'What were you referring to'
-    ) == 'You get'
-      message '*** Cannot run PrayerMat; no prayer mat in theurgy_supply_container'
+    unless DRCI.get_item?(@prayer_mat, @prayer_mat_container)
+      DRC.message("*** Cannot run PrayerMat; Cannot get prayer mat from prayer_mat_container, or from theurgy_supply_container.")
       @training_abilities.delete('PrayerMat')
       return
     end
-    fix_standing
-    bput("unroll #{@prayer_mat}",
-         'reverently lay your', 'need to be holding that first')
-
-    bput("kneel #{@prayer_mat}", 'You humbly kneel', 'You kneel in the water')
-    bput("kiss #{@prayer_mat}", 'You bend forward to kiss')
-
-    bput("get wine from my #{@theurgy_supply_container}",
-         'You get', 'What were you referring to')
-    bput("pour wine on #{@prayer_mat}",
-         'You quietly pour', 'Pour what', 'You set aside')
-    bput("put wine in my #{@theurgy_supply_container}",
-         'You put', 'What were you referring to')
-    fix_standing
+    DRC.fix_standing
+    DRC.bput("unroll #{@prayer_mat}", 'reverently lay your', 'need to be holding that first')
 
-    bput("roll #{@prayer_mat}",
-         'carefully gather up',
-         'need to be holding that first', 'not on the ground')
-    bput("put #{@prayer_mat} in my #{@prayer_mat_container}",
-         'You put', 'What were you referring to')
+    DRC.bput("kneel #{@prayer_mat}", 'You humbly kneel', 'You kneel in the water')
+    DRC.bput("kiss #{@prayer_mat}", 'You bend forward to kiss')
+
+    unless DRCI.get_item?("wine", @theurgy_supply_container)
```

Comment:
```suggestion
    if DRCI.get_item?("wine", @theurgy_supply_container)
        DRC.bput("pour wine on #{@prayer_mat}", 'You quietly pour', 'Pour what', 'You set aside')
        DRCTH.empty_cleric_hands(@theurgy_supply_container)
    else
        DRC.message("*** Finishing PrayerMat early; Cannot get wine from theurgy_supply_container")
        @training_abilities.delete('PrayerMat')
    end
```
and then the shared cleanup only needs to be done once

Reply:
How's this version?

Reply:
Looks good, except I don't think return on line 2721 should be there. Won't that short-circuit and not do the cleanup?

Reply:
Good catch! Thank you.